modules/potree/src/lib/potree-node-source.ts

```diff
@@ -0,0 +1,119 @@
+import {load} from '@loaders.gl/core';
+import {MeshGeometry} from '@loaders.gl/schema';
+import {DataSource, DataSourceProps, LoaderOptions, resolvePath} from '@loaders.gl/loader-utils';
+import {LASLoader} from '@loaders.gl/las';
+import {PotreeMetadata} from '../types/potree-metadata';
+import {POTreeNode} from '../parsers/parse-potree-hierarchy-chunk';
+import {PotreeHierarchyChunkLoader} from '../potree-hierarchy-chunk-loader';
+import {PotreeLoader} from '../potree-loader';
+
+export type PotreeNodesSourceProps = DataSourceProps & {
+  attributions?: string[];
+  potree?: {
+    loadOptions?: LoaderOptions; // PotreeLoaderOptions;
+    // TODO - add options here
+  };
+};
+
+/**
+ * A Potree data source
+ * @note Point cloud nodes tile source
+ */
+export class PotreeNodesSource extends DataSource {
+  url: string;
+  data: string | Blob;
+  props: PotreeNodesSourceProps;
+  metadata: PotreeMetadata | null = null;
+  root: POTreeNode | null = null;
+  isReady = false;
+
+  private initPromise: Promise<void> | null = null;
+
+  get isSupported(): boolean {
```

Comment:
No need for getters here. I think it is fine to call `isSupported()` as a function.
```suggestion
  isSupported(): boolean {
```

Reply:
Done